### PR TITLE
Remove deephaven-legacy and related code - 1st attempt, no reorg

### PR DESCRIPTION
**Pull request by jmao-denver**
Fixes: 2498

### CODE REVIEW

File: Integrations/README.md [hunk 1]
```diff
@@ -1,11 +0,0 @@
-### Building wheel with docker
```
Comment (rcaudy) [inline Integrations/README.md:1]: I'm concerned that we should just delete this file, rather than leaving it empty.

**Review by rcaudy — APPROVED** [review]
I would love to merge this, FWIW.

**Review by niloc132 — COMMENTED** [review]
Please update .github/workflows/docs-ci.yml to no longer copy these legacy docs over - check with @dsmmcken how we want to leave those legacy docs up, or if we just want to take them all down right away, how that rsync should be tweaked.

**Comment by niloc132** [comment]
I vote don't worry about the testing cleanup in Integrations/, we can do a followup for that.

**Review by devinrsmith — APPROVED** [review]
Happy to merge, see Colin's responses first.

**Review by jakemulf — DISMISSED** [review]

**Review by rcaudy — DISMISSED** [review]

**Review by rcaudy — APPROVED** [review]

**Pull request merged**